test/hotspot/jtreg/runtime/ErrorHandling/MachCodeFramesInErrorFile.java

```diff
@@ -127,9 +128,9 @@ private static void run(boolean crashInJava) throws Exception {
         // Extract hs_err_pid file.
         String hs_err_file = output.firstMatch("# *(\\S*hs_err_pid\\d+\\.log)", 1);
         if (hs_err_file == null) {
-            throw new RuntimeException("Did not find hs_err_pid file in output.\n" +
-                                       "stderr:\n" + output.getStderr() + "\n" +
-                                       "stdout:\n" + output.getStdout());
+            System.out.println("stdout:\n" + output.getStdout());
+            System.err.println("stderr:\n" + output.getStderr());
+            throw new RuntimeException("Did not find hs_err_pid file in output");
```

Comment:
Why this change (and a similar one at the end of the file)? This separates the reporting of a failed condition (i.e., could not find expected patten in output) from the context (i.e. contents of stderr and stdout) that assists in diagnosing why the condition may have failed.

Reply:
Also the test should really take advantage of output.shouldContain to check for the hs_err line, which in turn will report stdout and stderr on failure, and throw the exception.

Reply:
@dougxc, I added [an explanation](https://github.com/openjdk/jdk/pull/8094/files#r855362995) to the similar change at the end of the file. That was the one I discovered first. Here I made a similar change to prevent the same issue from occurring.

@dholmes-ora, I don't think output.shouldMatch is the best choice here because it involves doing the same operation twice. I believe I would end up looking for the pattern and throwing an error with shouldMatch, then looking for the pattern again and returning the String with firstMatch. I do like your idea of using more standard code however. I would like to replace the calls to println with `output.reportDiagnosticSummary()` if that is a reasonable compromise to your suggestion?

Reply:
@dougxc, I added [an explanation](https://github.com/openjdk/jdk/pull/8094/files#r855362995) to the similar change at the end of the file. That was the one I discovered first. Here I made a similar change to prevent the same issue from occurring.

@dholmes-ora, I don't think output.shouldMatch is the best choice here because it involves doing the same operation twice. I believe I would end up looking for the pattern and throwing an error with shouldMatch, then looking for the pattern again and returning the String with firstMatch. I do like your idea of using more standard code however. I would like to replace the calls to println with `output.reportDiagnosticSummary()` if that is a reasonable compromise to your suggestion?

Reply:
It is a test so I'm not overly concerned about scanning the output twice - though I agree it is a bit irksome to have to do that (new RFE: add `firstMatchOrFail`?). Using `reportDiagnosticSummary` is a reasonable compromise. Thanks.